File: CSharp/HtmlTest/Base/CloneTest_Base.cs
using System.IO;

using Microsoft.VisualStudio.TestTools.UnitTesting;
using HtmlWalker.Platform;
using System;

namespace HtmlTest.Base
{
    [TestClass]
    public class CloneTest_Base : CloneTest
    {
        public CloneTest_Base() : base("html", "Base") { }

        protected override WalkerPlatform Platform => throw new NotSupportedException();

        protected override void CreateExpected(FileInfo expected, FileInfo testInput, Options options = null) { }

        [TestMethod]
        public void TEmptyLines()
        {
            if (!CompareLines("", ""))
                Assert.Fail("Empty lines don't match");
        }

        [TestMethod]
        public void TSkipWhitespace()
        {
            if (!CompareLines(" x xx", "xx x "))
                Assert.Fail("Whitespace used to compare");
        }

        [TestMethod]
        public void TDifferentLengths0()
        {
            if (CompareLines("xxx", "xxxx"))
                Assert.Fail("Lines of different lenght match");
        }

        [TestMethod]
        public void TDifferentLengths1()
        {
            if (CompareLines("x xx", "xxxx"))
                Assert.Fail("Whitespace used for length");
        }
    }
}

File: Java/htmlwalker/src/test/java/htmltest/base/CloneTest_Base.java
package htmltest.base;

import static org.junit.Assert.fail;

import java.io.File;

import htmlwalker.platform.WalkerPlatform;

public class CloneTest_Base extends CloneTest
{
	private final static String TSTPATH = "Test";

    public CloneTest_Base() { super("html", "Base"); }

    @Override
    protected String testFolder() { return TSTPATH; }

    @Override
    protected boolean isXml() { return false; }

    @Override
    protected WalkerPlatform platform() { throw new UnsupportedOperationException(); }

    @Override
    protected void createExpected(File expected, File testInput, TestOptions options) { }

    public void tEmptyLines()
    {
        if (!compareLines("", "", true))
            fail("Empty lines don't match");
    }

    public void tSkipWhitespace()
    {
        if (!compareLines(" x xx", "xx x ", true))
            fail("Whitespace used to compare");
    }

    public void tDifferentLengths0()
    {
        if (compareLines("xxx", "xxxx", true))
            fail("Lines of different lenght match");
    }

    public void tDifferentLengths1()
    {
        if (compareLines("x xx", "xxxx", true))
            fail("Whitespace used for length");
    }
}
